cosign/verifier.go

```diff
@@ -0,0 +1,122 @@
+/*
+Copyright The Ratify Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cosign
+
+import (
+	"context"
+	"fmt"
+
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	ratify "github.com/ratify-project/ratify-go"
+	"github.com/sigstore/sigstore-go/pkg/bundle"
+	"github.com/sigstore/sigstore-go/pkg/root"
+	"github.com/sigstore/sigstore-go/pkg/verify"
+)
+
+const (
+	cosignVerifierType = "cosign"
+)
+
+// VerifierOptions contains the options for creating a new Cosign verifier.
+type VerifierOptions struct {
+	Name                  string
+	CosignVerifierOptions []verify.VerifierOption
+	TrustedMaterial       root.TrustedMaterial
+	MinBundleVersion      string
+	TrustedPublicOptions  map[string]trustedPolicyOption
+}
+
+// Verifier is a ratify.Verifier implementation that verifies Cosign
+// signatures.
+type Verifier struct {
+	name                 string
+	minBundleVersion     string
+	trustedPolicyOptions map[string]trustedPolicyOption
+	verifier             *verify.SignedEntityVerifier
+}
+
+// NewVerifier creates a new Cosign verifier.
+func NewVerifier(opts *VerifierOptions) (*Verifier, error) {
```

Comment:
does the created verifier support both keyless/keyed verification? What options would be different in terms of the scenario

Reply:
Options are configured in policy in trust material. And each component is decoupled with others.
There is no difference when create a new verifier here.